crypto/http/http_lib.c

```diff
@@ -21,11 +21,12 @@
  */
 
 int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
-                        char **ppath, int *pssl)
+                        int *pport_num, char **ppath, int *pssl)
 {
     char *p, *buf;
-    char *host;
-    const char *port = OSSL_HTTP_PORT;
+    char *host, *host_end;
+    const char *path, *port = OSSL_HTTP_PORT;
+    long portnum = 80;
```

Comment:
Why use a long and then return an int?

Reply:
This is because I use below `strtol()`, which is preferred over `a2int` etc. for its better error signaling.
A specific port range check is then needed anyway, an on success a cast to `int` can be done safely.